packages/nextjs/src/client/index.ts

```diff
@@ -86,13 +86,30 @@ function fixBrowserTracingIntegration(options: BrowserOptions): void {
   }
 }
 
+function isNewBrowserTracingIntegration(
+  integration: Integration,
+): integration is Integration & { options?: Parameters<typeof browserTracingIntegration>[0] } {
+  return !!integration.afterAllSetup && !!(integration as BrowserTracing).options;
+}
+
 function maybeUpdateBrowserTracingIntegration(integrations: Integration[]): Integration[] {
   const browserTracing = integrations.find(integration => integration.name === 'BrowserTracing');
+
+  if (!browserTracing) {
+    return integrations;
+  }
+
+  // If `browserTracingIntegration()` was added, we need to force-convert it to our custom one
+  if (isNewBrowserTracingIntegration(browserTracing)) {
```

Comment:
This is all pretty hacky 😬 but the best I could come up with! It should work well enough for 99% of cases.